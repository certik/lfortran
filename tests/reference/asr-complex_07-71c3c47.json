{
    "basename": "asr-complex_07-71c3c47",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/complex_07.f90",
    "infile_hash": "98ceed61d8a2eaf04519347bac51e261cf6167038a3238a7016f4f0e",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-complex_07-71c3c47.stdout",
    "stdout_hash": "cd6f67d9f01653d165c1ff16f67d32ba283accbad74aa519dfc5b67d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}