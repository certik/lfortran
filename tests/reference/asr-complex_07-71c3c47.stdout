(TranslationUnit (SymbolTable 1 {complex_07: (Program (SymbolTable 2 {abs: (ExternalSymbol 2 abs 4 abs lfortran_intrinsic_math [] abs Private), abs@cabs: (ExternalSymbol 2 abs@cabs 4 cabs lfortran_intrinsic_math [] cabs Private), img_part: (Variable 2 img_part Local (RealConstant   3.14000000000000012e+00 (Real 4 [])) (RealConstant   3.14000000000000012e+00 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), real_part: (Variable 2 real_part Local (IntegerConstant 42 (Integer 4 [])) (IntegerConstant 42 (Integer 4 [])) Parameter (Integer 4 []) Source Public Required .false.), x: (Variable 2 x Local (ComplexConstructor (Var 2 real_part) (RealConstant   0.00000000000000000e+00 (Real 4 [])) (Complex 4 []) ()) () Save (Complex 4 []) Source Public Required .false.), y: (Variable 2 y Local (ComplexConstructor (Var 2 img_part) (RealConstant   0.00000000000000000e+00 (Real 4 [])) (Complex 4 []) ()) () Save (Complex 4 []) Source Public Required .false.), z: (Variable 2 z Local (ComplexConstructor (Var 2 real_part) (Var 2 img_part) (Complex 4 []) ()) () Save (Complex 4 []) Source Public Required .false.)}) complex_07 [] [(If (RealCompare (FunctionCall 2 abs@cabs 2 abs [((ComplexBinOp (Var 2 x) Sub (ComplexConstructor (Var 2 real_part) (IntegerConstant 0 (Integer 4 [])) (Complex 4 []) (ComplexConstant   4.20000000000000000e+01   0.00000000000000000e+00 (Complex 4 []))) (Complex 4 []) ()))] (Real 4 []) () ()) Gt (RealConstant   1.00000000000000008e-05 (Real 4 [])) (Logical 4 []) ()) [(ErrorStop ())] []) (If (RealCompare (FunctionCall 2 abs@cabs 2 abs [((ComplexBinOp (Var 2 y) Sub (ComplexConstructor (Var 2 img_part) (IntegerConstant 0 (Integer 4 [])) (Complex 4 []) (ComplexConstant   3.14000000000000012e+00   0.00000000000000000e+00 (Complex 4 []))) (Complex 4 []) ()))] (Real 4 []) () ()) Gt (RealConstant   1.00000000000000008e-05 (Real 4 [])) (Logical 4 []) ()) [(ErrorStop ())] []) (If (RealCompare (FunctionCall 2 abs@cabs 2 abs [((ComplexBinOp (Var 2 z) Sub (ComplexConstructor (Var 2 real_part) (Var 2 img_part) (Complex 4 []) (ComplexConstant   4.20000000000000000e+01   3.14000000000000012e+00 (Complex 4 []))) (Complex 4 []) ()))] (Real 4 []) () ()) Gt (RealConstant   1.00000000000000008e-05 (Real 4 [])) (Logical 4 []) ()) [(ErrorStop ())] []) (Print () [(Var 2 x) (Var 2 y) (Var 2 z)] () ())]), iso_c_binding: (IntrinsicModule lfortran_intrinsic_iso_c_binding), iso_fortran_env: (IntrinsicModule lfortran_intrinsic_iso_fortran_env), lfortran_intrinsic_builtin: (IntrinsicModule lfortran_intrinsic_builtin), lfortran_intrinsic_math: (IntrinsicModule lfortran_intrinsic_math)}) [])
